### PR TITLE
refactor loadClass method

### PR DESCRIPTION
**Pull request by wanghbxxxx**
extract method ：cacheWrapperClass
extract method ：cacheAdaptiveClass
extract method ：cacheActivateClass
extract method ：saveInExtensionClasses
extract method ：cacheNames

(names != null && names.length > 0)，is always true, remove it

## What is the purpose of the change

XXXXX

## Brief changelog

XXXXX

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java [hunk 1]
```diff
@@ -714,33 +703,87 @@ private void loadClass(Map<String, Class<?>> extensionClasses, java.net.URL reso
...
-                        cachedNames.put(clazz, n);
-                    }
-                    Class<?> c = extensionClasses.get(n);
-                    if (c == null) {
-                        extensionClasses.put(n, clazz);
-                    } else if (c != clazz) {
-                        throw new IllegalStateException("Duplicate extension " + type.getName() + " name " + n + " on " + c.getName() + " and " + clazz.getName());
-                    }
-                }
+            cacheActivateClass(clazz, names[0]);
```
Comment (beiwei30) [inline dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java]: We should use defensive programming to judge **names** is not either null or empty, plus, I suggest we should loop **names** once and only once, see below:


```suggestion
            String[] names = NAME_SEPARATOR.split(name);
            if (ArrayUtils.isNotEmpty(names)) {
                cacheActivateClass(clazz, names[0]);
                for (String n : names) {
                    cacheName(clazz, n);
                    saveInExtensionClass(extensionClasses, clazz, name);
                }
            }
```

File: dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java [hunk 2]
```diff
@@ -714,33 +704,87 @@ private void loadClass(Map<String, Class<?>> extensionClasses, java.net.URL reso
...
             }
         }
     }
 
+    /**
+     * cache name
+     */
+    private void cacheName(Class<?> clazz, String name) {
+        if (!cachedNames.containsKey(clazz)) {
+            cachedNames.put(clazz, name);
```
[the remark this answers is not in the document]
Comment (wanghbxxxx) [reply dubbo-common/src/main/java/org/apache/dubbo/common/extension/ExtensionLoader.java]: thank you

**Comment by beiwei30** [comment]
all settled? can I review again?

**Comment by beiwei30** [comment]
@satansk do you have any further review comment?

**Comment by wanghbxxxx** [comment]
> all settled? can I review again?

im new to github, i dont know where comes this error: all checks have failed, please help me to deal with it. 
thankyou

**Comment by satansk** [comment]
@beiwei30 Nope :)

**Pull request merged**